contentcuration/contentcuration/viewsets/channel.py

```diff
@@ -505,26 +507,37 @@ def get_queryset(self):
 
     def annotate_queryset(self, queryset):
         queryset = super(AdminChannelViewSet, self).annotate_queryset(queryset)
-
-        editor_query = (
-            User.objects.filter(editable_channels__id=OuterRef("id"))
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        viewers_query = (
-            User.objects.filter(view_only_channels__id=OuterRef("id"))
-            .values_list("id", flat=True)
-            .distinct()
-        )
-        file_query = (
-            File.objects.filter(contentnode__tree_id=OuterRef("main_tree__tree_id"))
-            .values("checksum", "file_size")
-            .distinct()
-        )
-        queryset = queryset.annotate(
-            editors_count=SQCount(editor_query, field="id"),
-            viewers_count=SQCount(viewers_query, field="id"),
-            size=SQSum(file_query, field="file_size"),
-        )
-
         return queryset
+
+    def consolidate(self, items, queryset):
+        if items:
+            for item_channel in items:
+                metadata = self.get_or_cache_channel_metadata(
+                    item_channel["id"], item_channel["main_tree__tree_id"]
+                )
+                if metadata == DEFERRED_FLAG:
+                    item_channel["size"] = item_channel["editors_count"] = item_channel[
+                        "editors_count"
+                    ] = DEFERRED_FLAG
+                else:
+                    item_channel["size"] = metadata["SIZE"]
+                    item_channel["editors_count"] = metadata["EDITORS"]
+                    item_channel["viewers_count"] = metadata["VIEWERS"]
+        return items
+
+    def get_or_cache_channel_metadata(self, channel, tree_id):
+        key = "channel_metadata_{}".format(channel)
+        metadata = cache.get(key)
+        if metadata is None:
+            cache_channel_metadata_task.delay(channel, tree_id)
+            # from contentcuration.utils.channel import cache_channel_size
+            # cache_channel_size(channel, tree_id)
+            return DEFERRED_FLAG
+        else:
+            if "SIZE" in metadata:
+                # do we need to add a stale strategy here or keys
+                # will be invalidated when the channel of its nodes changes
+                # or just after they expire?
+                return metadata
+            else:
+                return DEFERRED_FLAG
```

Comment:
Perhaps we need to add an extra list endpoint here for fetching of deferred data by channel id?

```
@action(detail=False, methods=['get'])
def deferred_data(self, request):
    ids = request.GET.get("id__in")
    if not ids:
        raise ValidationError("id__in GET parameter is required")
    output = []
    for channel_id in ids:
        result = { "id": channel_id }
        result.update(self.get_metadata_for_channel(channel_id))
   return Response(output)
```
Where `get_metadata_for_channel` returns something like you make in the `consolidate` method, but does not kick off new tasks if deferred is returned.

Reply:
ah, yes, I thought of this but forgot to add the comment in the PR, for the frontend this could be needed but I didn't know if the idea was reusing the same api used to request the admin table, as all the data will be collected together instead of channel by channel. 
I'll add if this could be useful